site/docs/concepts/architecture/system-architecture.md

```diff
@@ -24,14 +24,15 @@ The architecture is divided into two main planes:
    - Rate Limit Service manages token-based rate limiting
 
 The Control Plane configures the Data Plane through several paths:
-- AI Gateway Controller configures the External Processor and Rate Limit Service
-- Envoy Gateway Controller configures Envoy Proxy through xDS
+- AI Gateway Controller configures the External Processor as well as fine-tuning xDS configurations via Envoy Gateway extension server machanism.
+- Envoy Gateway Controller configures Envoy Proxy through xDS as well as Rate Limit Service.
 
 ```mermaid
 graph TB
     subgraph "Control Plane"
         K8s[Kubernetes API Server] --> Controller[AI Gateway Controller]
-        Controller --> EGController[Envoy Gateway Controller]
+        Controller <-->|Translate
+                        xDS fine-tune| EGController[Envoy Gateway Controller]
```

Comment:
intentional new line between Translate and xds as otherwise they won't fit in!

Reply:
let me try

Reply:
My best https://github.com/envoyproxy/ai-gateway/pull/677/commits/713ecff6da417fffb9034d9bfcee6c58bd342067

<img width="662" alt="image" src="https://github.com/user-attachments/assets/13839a15-0dc9-4ab6-a44d-4cb52830d761" />

Reply:
thanks !